torch/distributed/nn/api/remote_module.py

```diff
@@ -544,8 +559,8 @@ def _remote_module_reducer(remote_module):
         elif k not in _REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING:  # type: ignore[attr-defined]
             print(
                 "The new attribute ``{}`` of RemoteModule is ignored during RPC pickling. "
-                "To pickle this attribute, it must be either in ``_REMOTE_MODULE_PICKLED_ATTRIBUTES`` or "
-                "``_REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING``.".format(k),
+                "To pickle this attribute, please add it to ``_REMOTE_MODULE_PICKLED_ATTRIBUTES``. "
+                "Otherwise, please explicitly add it to ``_REMOTE_MODULE_ATTRIBUTES_IGNORE_FOR_PICKLING``.".format(k),
```

Comment:
Totally correct. Currently we cannot prevent users from adding a new attribute dynamically, unless we add this sanity check to each method.

However, I tried to add a sanity check to the methods that will be installed via torch script. It's still hard to write a test to capture the exception w/o hitting timeout. Since this should rarely happen, I don't want to add more complexity here.